File: Easy/Count pairs with given sum/count-pairs-with-given-sum.py
#User function Template for python3

class Solution:
    def getPairsCount(self, arr, n, k):
        # code here
        cnt = 0;
        m = {}
        for i in range(n):
            b= k - arr[i]
            if b in m:
                cnt+=m[b]
            m[arr[i]] = m.get(arr[i], 0) + 1
            
        return cnt


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    tc = int(input())
    while tc > 0:
        n, k = list(map(int, input().strip().split()))
        arr = list(map(int, input().strip().split()))
        ob = Solution()
        ans = ob.getPairsCount(arr, n, k)
        print(ans)
        tc -= 1

# } Driver Code Ends

File: Difficulty: Easy/Count pairs with given sum/count-pairs-with-given-sum.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution {
  public:
    int countPairs(vector<int> &arr, int target) {
        // Code here
         sort(arr.begin(),arr.end());
         unordered_map<int,int> mp;
         int cnt=0;
         for(int i=0;i<arr.size();i++){
              int d=target-arr[i];
              if(mp.find(d)!=mp.end()){
                  cnt+=mp[d];
              }
              mp[arr[i]]++;
         }
         return cnt;
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    cin.ignore();
    while (t--) {
        vector<int> arr;
        string input;
        getline(cin, input);
        stringstream ss(input);
        int number;
        while (ss >> number) {
            arr.push_back(number);
        }
        int target;
        cin >> target;
        cin.ignore();
        Solution ob;
        int res = ob.countPairs(arr, target);

        cout << res << endl << "~" << endl;
    }
    return 0;
}
// } Driver Code Ends